sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -396,8 +396,9 @@ case class AlterTableChangeColumnCommand(
     val newDataSchema = table.dataSchema.fields.map { field =>
       if (field.name == originColumn.name) {
         // Create a new column from the origin column with the new comment.
+        val newField = field.copy(dataType = newColumn.dataType)
```

Comment:
Lines 359-367: the comment needs to be updated to be clear we now allow changing collation.

Side note: it gives the hive-style syntax for ALTER COLUMN but that's only one of the two syntaxes, see:
https://github.com/apache/spark/blob/master/sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4#L130

The other one is the one (partially) documented and the more capable/preferred one: https://spark.apache.org/docs/3.5.1/sql-ref-syntax-ddl-alter-table.html#parameters-4

Reply:
Updated the comment, please check.